composer/trainer/__init__.py

```diff
@@ -2,72 +2,66 @@
 
 """Train models!
 
-The trainer supports models with :class:`ComposerModel` instances.
-The :class:`Trainer` is highly customizable and can support a wide variety of workloads.
+The trainer supports models with :class:`~composer.models.base.ComposerModel` instances.
+The :class:`~composer.trainer.trainer.Trainer` is highly customizable and can
+support a wide variety of workloads.
 
-Examples
+Example
 --------
 
-.. code-block:: python
-
-    # Setup dependencies
-    from composer.datasets import MNISTDatasetHparams
-    from composer.models.mnist import MnistClassifierHparams
-    model = MnistClassifierHparams(num_classes=10).initialize_objeect()
-    train_dataloader = DataLoader(
-        datasets.MNIST('~/datasets/', train=True, transform=transforms.ToTensor(), download=True),
-        drop_last=True,
-        shuffle=True,
-        batch_size=256,
-    )
-
-    eval_dataloader = DataLoader(
-        datasets.MNIST('~/datasets/', train=True, transform=transforms.ToTensor(), download=True),
-        drop_last=False,
-        shuffle=False,
-        batch_size=256,
-    )
-
-
-.. code-block:: python
-
-    # Create a trainer that will checkpoint every epoch
-    # and train the model
-    trainer = Trainer(model=model,
-                      train_dataloader=train_dataloader,
-                      eval_dataloader=eval_dataloader,
-                      max_duration="50ep",
-                      train_batch_size=128,
-                      eval_batch_size=128,
-                      checkpoint_interval_unit="ep",
-                      checkpoint_folder="checkpoints",
-                      checkpoint_interval=1)
-    trainer.fit()
-
-
-.. code-block:: python
-
-    # Load a trainer from the saved checkpoint and resume training
-    trainer = Trainer(model=model,
-                      train_dataloader=train_dataloader,
-                      eval_dataloader=eval_dataloader,
-                      max_epochs=50,
-                      train_batch_size=128,
-                      eval_batch_size=128,
-                      checkpoint_filepath="checkpoints/first_checkpoint.pt")
-    trainer.fit()
-
-
-.. code-block:: python
-
-    from composer.trainer import TrainerHparamms
-
-    # Create a trainer from hparams and train train the model
-    trainer = hparams.initialize_object()
-    trainer.fit()
+.. doctest::
```

Comment:
Thoughts on examples to include here? I figured since we have so many tutorials about the trainer it's best to just include the most straightforward use case in the docs here (create a trainer, save a checkpoint, resume training later)

Could maybe add something with speedup methods too if anyone thinks it's relevant? Just trying to not make the example more verbose than it already is

Reply:
I would also be for documenting the Trainer on composer.trainer instead of composer.trainer.trainer -- to do this, rename `trainer.py` to `_trainer.py`, and then in `composer/trainer/__init__.py` add a line like this:

```python3

from composer.trainer._trainer import Trainer
Trainer.__module__ = __name__
```

Reply:
I'm good with what you have here, but can you break it up into individual Examples? And these should probalby be in the `Trainer` docstrings and not module levle right?

Reply:
Lol @hanlint @ravi-mosaicml you have conflicting opinions 
- First is to document the trainer in the top level module by renaming `trainer.py` to `_trainer.py`
- Second is to move the trainer docstraings to `trainer.py` and leave the module-level docs (relatively) empty


Thoughts?

Reply:
🤔 let's resolve offline!

Reply:
Agree with Hanlin's points here to break up the examples and move it to the init docstring. Because of how long the example is, it may make it slighlty cleaner to use ` ..testcode::` rather than `.. doctest::` (see https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html). But your call :)